remove_kernels.sh

```diff
@@ -0,0 +1,143 @@
+  # be removed only then - this is explained later on as a comment.
+  latest_eurolinux_kernel_path="$(find /boot -name ${latest_eurolinux_kernel_package/kernel/vmlinuz})"
+  grubby --set-default="${latest_eurolinux_kernel_path}"
+}
+
+update_grub() {
+  # Cover all distros and versions bootloader entries.
+  # This snippet has been copied from migrate2eurolinux.sh and it's about to
+  # be brainstormed what location/responsibility (script) suits it better,
+  # TODO: more EFI entries?
+  echo "Updating the GRUB2 bootloader..."
+  if [ -d /sys/firmware/efi ]; then
+    if [ -d /boot/efi/EFI/almalinux ]; then
+      grub2-mkconfig -o /boot/efi/EFI/almalinux/grub.cfg
+    elif [ -d /boot/efi/EFI/centos ]; then
+      grub2-mkconfig -o /boot/efi/EFI/centos/grub.cfg
+    else
+      grub2-mkconfig -o /boot/efi/EFI/redhat/grub.cfg
+    fi
+  else
+    grub2-mkconfig -o /boot/grub2/grub.cfg
+  fi
+}
+
+prepare_list_of_kernels_to_be_removed() {
+  # Consider several scenarios such non-EuroLinux kernels that come from e.g.
+  # ELRepo or other sources. What should be done with them is up to the user
+  # to decide via an answer or a parameter.
+  mapfile -t all_non_eurolinux_kernel_packages < \
+    <(printf -- '%s\n' "${installed_kernel_packages[@]}" | grep -Ev 'EuroLinux|Scientific' | sed 's@\ @\_@g')
+  mapfile -t migratable_distros_kernel_packages < \
+    <(printf -- '%s\n' ${all_non_eurolinux_kernel_packages[@]} | grep -E 'Red.Hat|CentOS|Oracle|Rocky|Alma' | sed 's@\ @\_@g')
+  echo "The following non-EuroLinux kernel packages are still remaining:"
+  printf -- '%s\n' "${all_non_eurolinux_kernel_packages[@]}"
+  echo "---"
+  if [ ${#migratable_distros_kernel_packages[@]} -gt 0 ]; then
+    echo "Among which these come from the systems the script supports migration from:"
+    printf -- '%s\n' "${migratable_distros_kernel_packages[@]}"
+    echo "---"
+  fi
+
+  echo "What should the script do:"
+  echo "1. leave all non-EuroLinux kernel packages as they are"
+  echo "2. remove all non-EuroLinux kernel packages"
+  if [ ${#migratable_distros_kernel_packages[@]} -gt 0 ]; then
+    echo "3. remove only the kernel packages that come from migratable distros"
+  fi
+  echo "Please type the number of the desired operation: "
+  read answer
+
+  case "$answer" in
+    1) 
+       echo "Leaving all kernel packages as they are."
+       exit 0
+       ;;
+    2) printf -- '%s\n' "${all_non_eurolinux_kernel_packages[@]%%|*}" > /root/kernel_packages_to_remove.txt ;;
+    3) 
+       if [ ${#migratable_distros_kernel_packages[@]} -gt 0 ]; then
+         printf -- '%s\n' "${migratable_distros_kernel_packages[@]%%|*}" > /root/kernel_packages_to_remove.txt
+       else
+         exit_message "Unknown answer: $answer."
+       fi
+       ;;
+    *) exit_message "Unknown answer: $answer."
+  esac
+}
+
+prepare_systemd_service() {
+  # Once there's a list of the kernel-related packages the user wants to 
```

Comment:
Tomek I knew that you will get that service 👍🏼 working ! 😸

Reply:
Turns out it wasn't working properly and I simplified it along with fixing the issue.